androidHyperskillApp/src/main/java/org/hyperskill/app/android/track_selection/details/fragment/TrackSelectionDetailsFragment.kt

```diff
@@ -105,6 +106,10 @@ class TrackSelectionDetailsFragment :
                 requireRouter().backTo(MainScreen)
                 mainScreenRouter.switch(StudyPlanScreen)
             }
+            ViewAction.NavigateTo.Home -> {
+                requireRouter().backTo(MainScreen)
+                mainScreenRouter.switch(HomeScreen)
```

Comment:
Will this work?
See old implementation https://github.com/hyperskill/mobile-app/blob/develop/androidHyperskillApp/src/main/java/org/hyperskill/app/android/placeholder_new_user/fragment/PlaceholderNewUserFragment.kt#L126

Reply:
Maybe we should somehow notify `AppFeature` from the shared code that new user selected track? If will do like this I think we don't need to handle any new view actions, existing AppFeature view actions will handle this.

iOS legacy code does that:
1. `AuthNewUserPlaceholderViewModel` notifies `AppViewModel` that home screen presentation is requested https://github.com/hyperskill/mobile-app/blob/develop/iosHyperskillApp/iosHyperskillApp/Sources/Modules/AuthNewUserPlaceholder/AuthNewUserPlaceholderViewModel.swift#L33
2. AppViewModel send corresponding view action for home screen navigation https://github.com/hyperskill/mobile-app/blob/develop/iosHyperskillApp/iosHyperskillApp/Sources/Modules/App/AppViewModel.swift#L103

Reply:
Also, see this registration flow from the feature task https://vyahhi.myjetbrains.com/youtrack/issue/ALTUX-2042

> Registration -> Track list -> Track page -> Project list (if user premium and track has project) -> Project page (if user premium and track has project) -> Study plan -> Project list (if project completed and track in progress) -> Study plan -> Track list (if track completed) -> [Cycle]

Could you please ask product team do we need to show project list after track is selected in scope of registration flow?

Reply:
> Will this work?
See old implementation https://github.com/hyperskill/mobile-app/blob/develop/androidHyperskillApp/src/main/java/org/hyperskill/app/android/placeholder_new_user/fragment/PlaceholderNewUserFragment.kt#L126

 You are right, the TrackSelection screen should become the root screen.

Reply:
>Maybe we should somehow notify AppFeature from the shared code that the new user selected track? If will do like this I think we don't need to handle any new view actions, existing AppFeature view actions will handle this.

We can do it via some global actions flow.
But, I think it is too complicated for this case.